pkg/autoscaler/autoscaler.go

```diff
@@ -240,25 +229,20 @@ func (a *Autoscaler) autoscaleFn(api userconfig.Resource) (func() error, error)
 
 		if request < currentReplicas {
 			downscaleStabilizationFloor = recs.maxSince(autoscalingSpec.DownscaleStabilizationPeriod)
+			if downscaleStabilizationFloor != nil {
+				downscaleStabilizationFloor = pointer.Int32(libmath.MaxInt32(*downscaleStabilizationFloor, 1))
```

Comment:
It there a reason that this should not be the opposite of the one below? e.g.

```go
downscaleStabilizationFloor = pointer.Int32(libmath.MinInt32(*downscaleStabilizationFloor, currentReplicas))
```

we can discuss this live